lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -75,7 +75,7 @@ class << self
         def new_client(config)
           config[:ssl_mode] = parse_ssl_mode(config[:ssl_mode]) if config[:ssl_mode]
           ::Trilogy.new(config)
-        rescue Trilogy::DatabaseError => error
+        rescue Trilogy::ProtocolError => error
```

Comment:
It looks like `ER_ACCESS_DENIED_ERROR = 1045` will be a `BaseConnectionError` instead of a `ProtocolError` (https://github.com/github/trilogy/pull/15/files#diff-ec6d3c2c7755944adec0effe7b0ca9f3340d4146ee5908c5273f26b54864bd6fR58), which I think means it's no longer getting translated the same way. We are probably missing a test for this case...

Reply:
Actually I don't think I've quite wrapped my head around what happens to the `BaseConnectionError`s here in general. They used to get translated into `ActiveRecord::ConnectionNotEstablished`, but now are they getting raised as is?

Reply:
Ah yes thank you! I'll add a test case and ensure that we're also rescuing connection errors when initializing the client here 👍

Reply:
Thank you!

Reply:
It looks like this method has explicit translation for a few more errors:

```
ER_BAD_DB_ERROR = 1049
ER_CONN_HOST_ERROR = 2003
ER_UNKNOWN_HOST_ERROR = 2005
```

and the trilogy PR has those all as `ProtocolError`. So I think we still need to rescue `ProtocolError` unless we're planning to turn those specific ones into `BaseConnectionError`. I don't know all these errors too well, so I don't have any strong opinions to offer. Rescuing both seems safest for now 🤷🏻‍♂️.

Reply:
Sounds good 👍 I think we may want `ER_CONN_HOST_ERROR` and `ER_UNKNOWN_HOST_ERROR` to show up as connection errors, but I don't think we even see these right now in Trilogy because host-related errors are being surfaced as `QueryError`s ..? Either way, `ER_BAD_DB_ERROR` is indeed a `ProtocolError`, so we def need to rescue that!